feat(config): Add hooks (#1218)
rustic-rs/rustic: config/README.md

| progress-interval | The interval at which progress indicators are shown.                              | "100ms"       | "1m"                     | RUSTIC_PROGRESS_INTERVAL |
| use-profile       | Profile or array of profiles to use. Allows to recursely use other profiles.      | Empty array   | "other" , ["2nd", "3rd"] | RUSTIC_USE_PROFILE       |

### Global Hooks `[global.hooks]`

> These external commands are before and after each commands, respectively.

Reviewer: ```suggestion
These external commands are run before and after each commands, respectively.
```

| Attribute   | Description                                      | Default Value | Example Value                 | Environment Variable |
| ----------- | ------------------------------------------------ | ------------- | ----------------------------- | -------------------- |

> | run-before  | Run the given command before execution           | not set       | "echo test", ["echo", "test"] |                      |
> | run-after   | Run the given command after successful execution | not set       | "echo test", ["echo", "test"] |                      |
> | run-failed  | Run the given command after failed execution     | not set       | "echo test", ["echo", "test"] |                      |
> | run-finally | Run the given command after every execution      | not set       | "echo test", ["echo", "test"] |                      |
> 

Reviewer: ```suggestion
| run-before  | Run the given command before execution           | not set       | ["echo test"] |                      |
| run-after   | Run the given command after successful execution | not set       | ["echo test"] |                      |
| run-failed  | Run the given command after failed execution     | not set       | ["echo test"] |                      |
| run-finally | Run the given command after every execution      | not set       | ["echo test"] |                      |

```

| progress-interval | The interval at which progress indicators are shown.                              | "100ms"       | "1m"                     | RUSTIC_PROGRESS_INTERVAL |
| use-profile       | Profile or array of profiles to use. Allows to recursely use other profiles.      | Empty array   | "other" , ["2nd", "3rd"] | RUSTIC_USE_PROFILE       |

### Global Hooks `[global.hooks]`

These external commands are before and after each commands, respectively.

> ** Note **: There are also repository hooks, which should be used for commands

Reviewer: ```suggestion
**Note**: There are also repository hooks, which should be used for commands
```

| time                  | Set the time saved in the snapshot.                                                     | Not set               |               |
| with-atime            | If true, includes file access time (atime) in the backup.                               | false                 |               |

### Backup Hooks `[backup.hooks]`

> These external commands are before and after each backup, respectively. Note
> that global hooks are run additionaly.

Reviewer: ```suggestion
These external commands are run before and after each backup, respectively.

**Note**: Global hooks are run additionally.
```

| Attribute   | Description                                      | Default Value | Example Value                 | Environment Variable |
| ----------- | ------------------------------------------------ | ------------- | ----------------------------- | -------------------- |

> | run-before  | Run the given command before execution           | not set       | "echo test", ["echo", "test"] |                      |
> | run-after   | Run the given command after successful execution | not set       | "echo test", ["echo", "test"] |                      |
> | run-failed  | Run the given command after failed execution     | not set       | "echo test", ["echo", "test"] |                      |
> | run-finally | Run the given command after every execution      | not set       | "echo test", ["echo", "test"] |                      |

Reviewer: ```suggestion
| run-before  | Run the given command before execution           | not set       | ["echo test"] |                      |
| run-after   | Run the given command after successful execution | not set       | ["echo test"] |                      |
| run-failed  | Run the given command after failed execution     | not set       | ["echo test"] |                      |
| run-finally | Run the given command after every execution      | not set       | ["echo test"] |                      |
```

### Backup Sources `[[backup.sources]]`

**Note**: All of the backup options mentioned before can also be used as
source-specific option and then only apply to this source.

> Source-specific hooks are called when backup up the defined source, additionally
> to global and backup hooks. only apply

Reviewer: ```suggestion
Source-specific hooks are called additionally to global and backup hooks
when backing up the defined source.

```

| time                  | Set the time saved in the snapshot.                                                     | Not set               |               |
| with-atime            | If true, includes file access time (atime) in the backup.                               | false                 |               |

### Backup Hooks `[backup.hooks]`

These external commands are before and after each backup, respectively.

** Note **: Global hooks and repository hooks are run additionaly.

See Global Hooks.

### Backup Sources `[[backup.sources]]`

**Note**: All of the backup options mentioned before can also be used as
source-specific option and then only apply to this source.

Source-specific hooks are called additionally to global, repository and backup

> hooks when backup up the defined source.

Reviewer: ```suggestion
hooks when backing up the defined source.
```
